### PR TITLE
Mock response object, unit tests

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
Proposing to mock the response object so that we can test the command line interface with unit tests.

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-2873 Add pagination to the QFC API endpoints](https://app.clickup.com/t/2192114/QF-2873)

**Comment by why-not-try-calmer** [comment]
Downgrading to Draft as proper unit tests will require more footwork.

File: qfieldcloud_sdk/cli.py [hunk 1]
```diff
@@ -112,6 +113,7 @@ def cli(
     """
     ctx.ensure_object(dict)
     ctx.obj["client"] = sdk.Client(url, verify_ssl, token=token)
+    ctx.obj["session"] = requests.Session()
```
Comment (why-not-try-calmer) [inline qfieldcloud_sdk/cli.py:116]: I am not sure session can be persisted between commands but there's nothing to lose in acting as if it is.

File: qfieldcloud_sdk/sdk.py [hunk 2]
```diff
@@ -18,7 +18,10 @@
 import requests
 from requests.models import Response
 
-__version__ = metadata.version("qfieldcloud_sdk")
+try:
+    __version__ = metadata.version("qfieldcloud_sdk")
+except metadata.PackageNotFoundError:
+    __version__ = "dev"
```
Comment (why-not-try-calmer) [inline qfieldcloud_sdk/sdk.py:24]: Here the idea is that fetching the version from git doesn't always work, so this snippet is the spiritual successor of VERSION:dev.

**Pull request merged**